src/core/utils/color/ColorValue.ts

```diff
@@ -0,0 +1,709 @@
+    );
+  }
+
+  /** Create an HslColor from this ColorValue */
+  public toHSL(): HslColor {
+    // internally h,s,l ranges are in 0.0 - 1.0
+    const col = this.colors;
+    col.r /= 255;
+    col.g /= 255;
+    col.b /= 255;
+    const max = Math.max(col.r, col.g, col.b);
+    const min = Math.min(col.r, col.g, col.b);
+
+    let hue = 0;
+    let saturation;
+    const lightness = (min + max) / 2.0;
+
+    if (min === max) {
+      saturation = 0;
+    } else {
+      const delta = max - min;
+      saturation =
+        lightness <= 0.5 ? delta / (max + min) : delta / (2 - max - min);
+      switch (max) {
+        case col.r:
+          hue = (col.g - col.b) / delta + (col.g < col.b ? 6 : 0);
+          break;
+        case col.g:
+          hue = (col.b - col.r) / delta + 2;
+          break;
+        case col.b:
+          hue = (col.r - col.g) / delta + 4;
+          break;
+      }
+      hue /= 6;
+    }
+
+    return {
+      h: Math.round(hue * 360),
+      s: Math.round(saturation * 100),
+      l: Math.round(lightness * 100),
+    };
+  }
+
+  /** Create an [[RgbColor]] from this ColorValue */
+  public toRGB(): RgbColor {
+    const { r, g, b } = ColorValue.getColors(this._tbgr);
+    return { r, g, b };
+  }
+
+  /** Create a ColorValue from an HsvColor transparency => 0-255 */
+  public static fromRGB(rgb: RgbColor, transparency = 0): ColorValue {
+    return ColorValue.from(rgb.r, rgb.g, rgb.b, transparency);
+  }
+
+  /** Create an [[HsvColor]] from this ColorValue */
+  public toHSV(): HsvColor {
+    const { r, g, b } = this.colors;
+    let min = r < g ? r : g;
+    if (b < min) {
+      min = b;
+    }
+
+    let max = r > g ? r : g;
+    if (b > max) {
+      max = b;
+    }
+
+    /* amount of "blackness" present */
+    const v = Math.floor((max / 255.0) * 100 + 0.5);
+    const deltaRgb = max - min;
+    const s = max !== 0.0 ? Math.floor((deltaRgb / max) * 100 + 0.5) : 0;
+    let h = 0;
+
+    if (s) {
+      const redDistance = (max - r) / deltaRgb;
+      const greenDistance = (max - g) / deltaRgb;
+      const blueDistance = (max - b) / deltaRgb;
+
+      let intermediateHue: number;
+      if (r === max) {
+        /* color between yellow & magenta */
+        intermediateHue = blueDistance - greenDistance;
+      } else if (g === max) {
+        /* color between cyan & yellow */
+        intermediateHue = 2.0 + redDistance - blueDistance;
+      } else {
+        /* color between magenta & cyan */
+        intermediateHue = 4.0 + greenDistance - redDistance;
+      }
+
+      /* intermediate hue is [0..6] */
+      intermediateHue *= 60;
+
+      if (intermediateHue < 0.0) {
+        intermediateHue += 360;
+      }
+
+      h = Math.floor(intermediateHue + 0.5);
+
+      if (h >= 360) {
+        h = 0;
+      }
+    } else {
+      h = 0;
+    }
+
+    return { h, s, v };
+  }
+
+  /** Create a ColorValue from an HsvColor */
+  public static fromHSV(hsv: HsvColor, transparency = 0): ColorValue {
```

Comment:
Not sure about having separate methods to get `ColorValue` and having constructor with only tbgr support. Could we have just constructor, which could accept any of the supported formats as input and create `ColorValue` object that way? Would be consistent between different color schemes and possibly easier to use.

Reply:
used with displaying values. see ColorValue.create for all supported input types.